### ISSUE TITLE
Add limits of time and memory usages when evaluating pipelines

### ISSUE THREAD

**Issue by weixuanfu2016** [rhiever/tpot#249]
The pipeline evaluations may spend too much time when processing a large data set or improper data (e.g. data for a regression problem was provided to the TPOT Classifier object, see example below). Adding limits of time and memory usages can discard the pipelines that use too many computational resources and accelerate the pipeline evaluation in TPOT.

## Example
from tpot import TPOT
from sklearn.cross_validation import train_test_split
from sklearn.datasets import make_regression
reg_data = make_regression(n_samples=10000, n_features=100, n_informative=10, n_targets=1, bias=0.0, effective_rank=None, tail_strength=0.5, noise=0.0, shuffle=True, coef=False, random_state=99)
X_train, X_test, y_train, y_test = train_test_split(reg_data[0], reg_data[1],train_size=0.75, test_size=0.25, random_state = 88)
pipeopt = TPOT(generations=5, 
               population_size=40,
               num_cv_folds=5, 
               random_state=42, 
               verbosity=2,
               disable_update_check=True)

pipeopt.fit(X_train, y_train)

## Expected result
GP progress gets stuck at 8% (19/240) for more than 2 hours

**Comment by rhiever** [comment]
Added to the 0.7 release milestone.

**Comment by mfeurer** [comment]
If you don't want to write that yourself, have a look at the [pynisher package](https://github.com/sfalkner/pynisher). It doesn't run on windows though.

**Comment by rhiever** [comment]
Thank you for the tip @mfeurer! I was checking that package out last week and noticed the non-Windows limitation as well. Have you run across any promising ideas for limiting evaluations similar to pynisher that will work on all platforms?

**Comment by mfeurer** [comment]
I don't know of any solution other than the pynisher or writing it yourself. If I knew of, I would use it. To learn more about this, you should ask @sfalkner who wrote the pynisher.

**Comment by rhiever** [comment]
OK, I will keep you posted if we come up with anything.

**Comment by rhiever** [comment]
@mfeurer: We recently merged a PR with a basic implementation of this functionality for Windows, Linux, and Mac. You can see it [here](https://github.com/rhiever/tpot/blob/development/tpot/decorators.py#L75). We're still cleaning it up, but it's a good start.
